packages/docs-nextra/scripts/wrap-doenet-editor.ts

```diff
@@ -0,0 +1,47 @@
+import { visit } from "unist-util-visit";
+import { Plugin } from "unified";
+
+import { Root as MdastRoot } from "mdast";
+// Importing this automatically imports all the types from `mdast-util-mdx-jsx`.
+import "mdast-util-mdx-jsx";
+
+/**
+ * Wraps codeblocks of type `doenet-editor` with a JSX element that renders the editor with the output.
+ */
+export const wrapDoenetEditor: Plugin<void[], MdastRoot, MdastRoot> =
+    function () {
+        return (tree) => {
+            visit(tree, (node) => {
+                type MdxJsxFlowElement = typeof node & {
+                    type: "mdxJsxFlowElement";
+                };
+                if (node?.type !== "code") {
+                    return;
+                }
+                if (node.lang !== "doenet-editor") {
+                    return;
+                }
+
+                console.log(
+                    "value for wrapped editor",
+                    JSON.stringify(node.value),
+                );
```

Comment:
This is going to make the logs look really ugly as hundreds of examples get integrated into the docs.

Reply:
Yes, this is only there until I figure out why the newlines are disappearing. They are still here with this `console.log`, but then they are gone in the `console.log` on line 21 of `packages/docs-nextra/components/doenet.tsx`. It's mysterious and basically makes triple backtick doenet-editor non-functional. I'm not sure where to look for this removal of whitespace and am hoping you have ideas!